C/primitive/elements/jets.c

```diff
@@ -122,32 +119,52 @@ static void issuanceEntropy(frameItem* dst, const assetIssuance* issuance) {
  *               NULL != issuance;
  */
 static void issuanceAssetAmt(frameItem* dst, const assetIssuance* issuance) {
-  if (writeBit(dst, NO_ISSUANCE != issuance->type && NONE != issuance->assetAmt.prefix)) {
-    if (!amt(dst, &issuance->assetAmt)) {
-      /* The 'amt' function has side-effects so we cannot simply 'assert(amt(..))' as it would be removed in non-debug builds. */
-      assert(false);
-    }
+  if (writeBit(dst, NO_ISSUANCE != issuance->type)) {
+    amt(dst, &issuance->assetAmt);
   } else {
     skipBits(dst, 258);
   }
 }
 
-/* Write an optional confidential issued/reissued asset amount from an 'assetIssuance' to the 'dst' frame, advancing the cursor.
+/* Write an optional confidential token amount from an 'assetIssuance' to the 'dst' frame, advancing the cursor.
  *
  * Precondition: '*dst' is a valid write frame for 259 more cells;
  *               NULL != issuance;
  */
 static void issuanceTokenAmt(frameItem* dst, const assetIssuance* issuance) {
-  if (writeBit(dst, NEW_ISSUANCE == issuance->type) && NONE != issuance->tokenAmt.prefix) {
-    if (!amt(dst, &issuance->tokenAmt)) {
-      /* The 'amt' function has side-effects so we cannot simply 'assert(amt(..))' as it would be removed in non-debug builds. */
-      assert(false);
-    }
+  if (writeBit(dst, NEW_ISSUANCE == issuance->type)) {
+    amt(dst, &issuance->tokenAmt);
   } else {
     skipBits(dst, 258);
   }
 }
 
+/* Write an optional confidential asset rangeproof from an 'assetIssuance' to the 'dst' frame, advancing the cursor 257 cells.
+ *
+ * Precondition: '*dst' is a valid write frame for 257 more cells;
+ *               NULL != issuance;
+ */
+static void issuanceAssetProof(frameItem* dst, const assetIssuance* issuance) {
+  if (writeBit(dst, NO_ISSUANCE != issuance->type && is_confidential(issuance->assetAmt.prefix))) {
```

Comment:
Here and below, it looks like you may be inconsistently handling the non-confidential case.  It seems sometimes you write none in an optional type, but in other places you write a hash of the empty string.

Reply:
This now consistently returns the hash of the empty string.